app/controllers/references_controller.rb

```diff
@@ -35,7 +35,20 @@ def destroy
     end
   end
 
+  # GET /username/references/1/edit_abstract
+  def edit_abstract
```

Comment:
so should all edit functionality go through this action?  I guess I could also have the form pop up purely through javascript and not use ajax at all.  Let's talk through this since I clearly do not understand routing well enough. :-/